sound/soc/sof/intel/hda.c

```diff
@@ -808,17 +809,25 @@ static irqreturn_t hda_dsp_interrupt_thread(int irq, void *context)
 	struct sof_intel_hda_dev *hdev = sdev->pdata->hw_pdata;
 
 	/* deal with streams and controller first */
-	if (hda_dsp_check_stream_irq(sdev))
+	if (hda_dsp_check_stream_irq(sdev)) {
 		hda_dsp_stream_threaded_handler(irq, sdev);
+		trace_sof_intel_hda_irq_stream(sdev);
```

Comment:
maybe it's a nit-pick, but I would first trace the event, then deal with the event rather than throw the trace after possible processing.

One valid objection is that this would only reliably work for stream interrupts, the rest is always delayed by the handling of events with a higher priority (defined by the order of processing)

Reply:
This has been changed as per Marc's request.